### DISCUSSION TITLE
✔ Documentation doesn't clearly explain how to get started quickly

### DISCUSSION THREAD

**Discussion by kevinlinxc** [IndiePython/nodezator#38] [General]
There are a few problems I have with the documentation. 

1. This picture is nice, but an actual practical example directory would be nice:
![image](https://user-images.githubusercontent.com/44324811/205382623-4209ae44-e10e-4a1f-bc3c-0d229ff81de8.png)
For example, using this: https://ascii-tree-generator.com/, you could make an example tree of what the directory should look like.

2. How do I see the node made in the first tutorial?
We learn how to make a node like so:
![image](https://user-images.githubusercontent.com/44324811/205382849-0daf88ca-f073-44b3-b4e7-c22c9514ab0c.png)

, but there aren't instructions on how to load this node in and see it in the nodezator UI.

I hope you don't misunderstand, I believe this documentation I am looking for exists, but I think most users would prefer a quick start where they can write a node and see it in their window.

**Comment by KennedyRichard** [comment]
Hello, Mr. Kevin,

Those are both clever observations. Addressing each point...

> This picture is nice, but an actual practical example directory would be nice

Now that I think of it, the tree structure format you proposed really seems simpler and more efficient to convey the structure of a node pack. Here's a quick sketch I made taking your feedback into account. Please, let me know what you think of it. All I needed was html, didn't even need .svg like the current image, which also makes things simpler.

![node_pack_repr](https://user-images.githubusercontent.com/18728298/205448848-e2040893-739f-40d3-bbd2-03c0b0df1d1d.png)

[node_pack_repr.html.zip](https://github.com/IndiePython/nodezator/files/10146815/node_pack_repr.html.zip)



> How do I see the node made in the first tutorial? [...] I think most users would prefer a quick start where they can write a node and see it in their window.

You are right. The manual, despite having all the basic information needed to operate Nodezator, was a bit rushed and has a lot to improve. The information you need is on [this page on the manual](https://manual.nodezator.com/ch-loading-nodes.html).

Also, I agree that this information should be present in the first page of the manual, right after teaching people how to create their first node. I'll schedule some time to improve this ASAP.

I'll probably do this: add a tiny summarized version of the instructions from the chapter I linked above in the first chapter and also add a link to the full chapter on loading nodes. This way people learning how to create nodes will be able to load their nodes right away.

------

I'm not sure I'll be able to take a look into your post on the discussions today by it should be answered by Monday.

Have a nice weekend and thank you for your feedback.

**Comment by kevinlinxc** [comment]
Hi Kennedy,

I really like your proposed tree picture.

Thank you for being so responsive to feedback. I believe what you are making is great and has the potential to help many people.

**Comment by KennedyRichard** [comment]
Giving the attention the feedback of the users/contributors/community deserve is my duty and pleasure. Everyone benefits in the end. Also thank you for the kind words regarding the work put in Nodezator!

---

You can ignore this part of the message. I'm just quoting [your suggestion from discord](https://discord.com/channels/980193047000272967/980193047461630113/1048378415973142640) so that I don't forget to implement it when I perform the other changes mentioned above.

> I think a button that said "import nodes" or file-import nodes would be best instead of graph-> select node packs

**Comment by KennedyRichard** [comment]
Hello, Mr. Kevin

I took the liberty of turning this issue into a discussion. Don't worry, this doesn't change my mind about it, it is just a matter of organization.

In the IndiePython project (and in my general development practice), we don't call something an issue unless it causes something to break or prevent something from working as supposed to. For instance, check the [closed issues for Nodezator](https://github.com/IndiePython/nodezator/issues?q=is%3Aissue+is%3Aclosed). Most of it is critical stuff that makes the app crash and disrupt the workflow in the app.

This is just so that tasks can be properly scheduled based on urgency. When every single topic gets crammed in Issues, the urgency of the matters is often lost or difficult to assess.

Don't let this discourage you of posting stuff on Issues, though. We can always assess the real urgency later and either keep it as an issue or convert it into a discussion.

By the way, I didn't push into Nodezator yet, but I already changed the image representing the anatomy of a node pack according to what we discussed. Now I just need to update the online manual as well and also implement your suggestion of changing the saying in the menubar about loading nodes. It'll probably go from **Graph > Select node packs** to **Graph > load nodes** (rather than **Graph > import nodes** though). I'm actually still pondering.

Have a nice week

**Comment by KennedyRichard** [comment]
Finished adding your suggestions today. Changed the anatomy of node pack image and also included a new subsection in the first chapter summarizing how to load the node and pointing the the chapter that teaches how to do so. The changes were already pushed here and uploaded to pypi.org and can be accessed by upgrading Nodezator to the latest version with

```bash
pip install --upgrade nodezator
```

Thank you again for the feedback.

**Comment by KennedyRichard** [comment]
Forgot to close the discussion. So I'm doing it now.

Peace
